### PR TITLE
Added new post-processing scripts for PyGRB

### PR DESCRIPTION
**Pull request by duncanmmacleod**
This PR adds new executables for the PyGRB workflow to go with @idorrington92's new `pycbc_multi_inspiral` code

- `pycbc_grb_trig_combiner`
- `pycbc_grb_trig_cluster`

### CODE REVIEW

**Comment by spxiwh** [comment]
@idorrington92 and/or @pannarale should review this.

Please read through the code and indicate that you can understand the code that is written and follow what it is doing (and that what it is doing is right). Where it is not clear please ask for in-line comments (or whatever needed) to improve matters.

**Comment by idorrington92** [comment]
Sure, I'll do it today.

File: bin/pygrb/pycbc_grb_trig_combiner [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+    required=True,
+    metavar="TRIGGER FILE",
+    help="read in listed trigger files",
+)
+
+args = parser.parse_args()
+
+vprint = print if args.verbose else str
+
+vprint("-- Hello")
```
Comment (idorrington92) [inline bin/pygrb/pycbc_grb_trig_combiner]: Don't think we need 'hello'

File: bin/pygrb/pycbc_grb_trig_cluster [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+                        compression_opts=old.compression_opts,
+                    )
+                    bar.update()
+            for ifo in ifos:
+                idx = ifo_index[ifo]
+                for old in h5in[ifo].values():
+                    if isinstance(old, h5py.Dataset):
+                        h5out.create_dataset(
+                            old.name,
+                            data=old[:][idx],
```
Comment (idorrington92) [inline bin/pygrb/pycbc_grb_trig_cluster]: I think idx should be include.

Comment (duncanmmacleod) [reply bin/pygrb/pycbc_grb_trig_cluster]: @idorrington92, no, `include` is the index-array of network events to keep, `idx` here is the equivalent index-array of single-detector events to keep, so this is correct.

File: bin/pygrb/pycbc_grb_trig_combiner (hunk 1 above)
Comment (duncanmmacleod) [reply bin/pygrb/pycbc_grb_trig_combiner]: There's nothing wrong with a polite welcome...

**Comment by idorrington92** [comment]
Ok, I'm running on a different GRB, and this line is crashing. I'll keep
investigating.

**Comment by spxiwh** [comment]
Travis is failing:
```
>> [Fri Jun 28 16:03:40 UTC 2019] running /home/travis/virtualenv/python2.7.15/bin/pycbc_grb_inj_combiner --help
    FAILED!
---------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/virtualenv/python2.7.15/bin/pycbc_grb_inj_combiner", line 33, in <module>
    import tqdm
ImportError: No module named tqdm
---------------------------------------------------------
```

File: bin/pygrb/pycbc_grb_inj_combiner [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+    """Returns `True` if this file can be used to analysis the given inclination
+    """
+    match = INJSTRING_REGEX.search(filename)
+    if match and float(match.groups()[0]) >= inclination:
+        return True
+    return False
+
+
+def theta(h5injgroup):
+    """Calculate the theta angle column for an injection group
```
Comment (spxiwh) [inline bin/pygrb/pycbc_grb_inj_combiner]: I think there's better versions of this function in LAL (including the non-leading terms in L). We might want to switch to a better function if there is one. Worth checking with Seb/Patricia/Harald, but doesn't hold up merging this.

**Comment by idorrington92** [comment]
Inj finder returns the same injections repeatedly. Guessing there is a bug where it does not loop over the files correctly or something like that. I'll try to dig into it if I have time, otherwise it'll have to wait for @duncanmmacleod  to return from his holiday.

File: bin/pygrb/pycbc_grb_inj_combiner (hunk 3 above)
Comment (spxiwh) [reply bin/pygrb/pycbc_grb_inj_combiner]: Would be nice to have the conversion function in here though:
https://github.com/gwastro/pycbc/blob/master/pycbc/conversions.py

**Comment by spxiwh** [comment]
The Travis failure is because a new dependency (tqdm) is being added, but it has not been declared. I should also ask why the new dependency is needed, but I'm not too worried about that (other than a minor concern that it makes the code hard to understand, because `tqdm.tqdm` gives me *no* idea what is going on)

File: bin/pygrb/pycbc_grb_inj_combiner (hunk 3 above)
Comment (duncanmmacleod) [reply bin/pygrb/pycbc_grb_inj_combiner]: I am not confident that I understand the function and its use cases properly enough to translate this into that module. I would rather leave it here for now, and pass the buck to other pycbc devs to move this upstream and modify it to fit appropriately.

**Comment by duncanmmacleod** [comment]
> The Travis failure is because a new dependency (tqdm) is being added, but it has not been declared. I should also ask why the new dependency is needed, but I'm not too worried about that (other than a minor concern that it makes the code hard to understand, because `tqdm.tqdm` gives me _no_ idea what is going on)

@spxiwh, `tqdm` just provides the fancy progress bars, and is fairly ubiquitous in python libraries these days (it's already in the release virtualenvs under `/cvmfs`). Should this be a first-class dependency in `setup.py`, or just an optional extra in `requirements.txt`?

**Comment by spxiwh** [comment]
@duncanmmacleod No strong preference on my part. Let's put it in both `setup.py` and `requirements.txt` for now.

**Comment by duncanmmacleod** [comment]
ok, @spxiwh, I think this is ready for final review/merge, thanks.

**Pull request merged**